libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.services.android.navigation.v5.navigation.camera
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.core.constants.Constants
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import java.util.ArrayList
+
+/**
+ * The default camera used by [com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation].
+ */
+open class SimpleCamera : Camera() {
+
+    companion object {
+        protected const val DEFAULT_TILT = 50
+        protected const val DEFAULT_ZOOM = 15.0
+    }
+
+    private var routeCoordinates: List<Point> = ArrayList()
+    private var initialRoute: DirectionsRoute? = null
+
+    override fun tilt(routeInformation: RouteInformation): Double {
+        return DEFAULT_TILT.toDouble()
+    }
+
+    override fun zoom(routeInformation: RouteInformation): Double {
+        return DEFAULT_ZOOM
+    }
+
+    override fun overview(routeInformation: RouteInformation): List<Point> {
+        if (routeCoordinates.isEmpty()) {
+            buildRouteCoordinatesFromRouteData(routeInformation)
+        }
+        return routeCoordinates
+    }
+
+    private fun buildRouteCoordinatesFromRouteData(routeInformation: RouteInformation) {
+        if (routeInformation.route != null) {
+            setupLineStringAndBearing(routeInformation.route)
+        } else if (routeInformation.routeProgress != null) {
+            setupLineStringAndBearing(routeInformation.routeProgress.directionsRoute())
+        }
+    }
+
```

Comment:
The function buildRouteCoordinatesFromRouteData() should be rewritten like so:

 ifNonNull(routeInformation.route) {route ->
                 setupLineStringAndBearing(route)
} ?: ifNonNull (routeInformation.routeProgress) {routeProgress->
                 setupLineStringAndBearing(routeProgress.directionsRoute())
 }